src/main/java/de/tum/cit/aet/artemis/assessment/service/AssessmentService.java

```diff
@@ -283,6 +287,9 @@ public Result saveManualAssessment(final Submission submission, final List<Feedb
         User user = userRepository.getUser();
         result.setAssessor(user);
 
+        // long feedback text is deleted as it otherwise causes duplicate entries errors and will be saved again with resultRepository.save(result)
+        resultService.deleteLongFeedback(feedbackList);
```

Comment:
Do we do this deletion here, because we only want to associate the long feedbacks with the newest result and not any old result?
Do we really want to break the old results so students can no longer look at them?

Reply:
Yes, we need this to avoid a duplicate entry error in the LongFeedbackRepository when saving a result in the ResultRepository. This approach doesn’t break the previous results, as it only applies when saving manual feedback after the due date. This means we aren’t working with multiple results; we’re only modifying a single result, which we overwrite each time.
![image](https://github.com/user-attachments/assets/2fe10305-0d06-4e24-a353-84abd6a58307)